app/funcs/csv_reader.py

```
@@ -80,11 +80,65 @@ def read_csv(self, file: Path, **kwargs) -> Generator[Dict[int, Dict], None, Non
+        try:
+            # Use scan_parquet for lazy evaluation
+            lazy_df = pl.scan_parquet(file, **kwargs)
+            
+            # Collect the lazy frame and then iterate over rows
+            df = lazy_df.collect()
+            
+            # Use iter_rows to get each row as a dictionary
+            for row in df.iter_rows(named=True):
+                yield row
+                
+        except Exception as e:
```

Ruff (0.14.7)

Do not catch blind exception: `Exception`

(BLE001)